pkgs/applications/window-managers/hyprwm/hyprland/plugins.nix

```
@@ -0,0 +1,51 @@
+{ lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, gcc13Stdenv
+, hyprland
+}:
```

```suggestion
{ lib
, fetchFromGitHub
, pkg-config
, gcc13Stdenv
, hyprland
, callPackage
}:
```

How about this? This way each plugin can specify its dependencies like a normal package would.

---

Not sure to follow you on this. You would create one file for each plugin then and add their dependencies?

---

Well, you'd make a plugin like this, in the same file:
```nix
hyprbars = callPackage (
{ cmake
, pixman
, cairo}:
mkHyprlandPlugin {
  pname = "hyprbars";
  version = "0.1";

  src = fetchFromGitHub ...;

  buildInputs = [
    cairo
    pixman
  ];

  nativeBuildInputs = [cmake];

  # ...
}) {};
```
Sure, it looks a little weird, but we can integrate the `callPackage` call inside `mkHyprlandPlugin`.

---

Ok I fixed it. @fufexan can you have a look please?